### PR TITLE
Fix parsing of csv template files

### PR DESCRIPTION
**Pull request by gmischler**
As of issue #215, here's the pull request:

Values of csv files are now converted by position, instead of content, so that no kind of text can cause an error.
I also added a converter to make sure that hex and oct color values are accepted for foreground and background in csv files.
Updated the test data to include an entry with only digits as text, to check for the regression (adequately failed before the code fix).
Updated the documentation and examples to reflect the changes and match the test data again.
While I was at it, I also included the recently improved multiline functionality in the documentation examples and in the tests.

Looks like I forgot to actually remove `util.try_to_type()` though. Is that used anywhere else?

### CODE REVIEW

File: fpdf/template.py [hunk 1]
```diff
@@ -74,26 +74,37 @@ def load_elements(self, elements):
...
-            "foreground",
-            "background",
-            "align",
-            "text",
-            "priority",
-            "multiline",
+        handlers = (
+            ("name", str.strip),
+            ("type", str.strip),
+            ("x1", float),
```
Comment (alexp1917) [inline fpdf/template.py:93]: i like this approach. "in the name of deterministic behavior" why strip the fields? but this is a good place to put the actual conversions, and less lines of code than what i was considering when i put the util function in.

File: fpdf/template.py [hunk 2]
```diff
@@ -74,26 +74,37 @@ def load_elements(self, elements):
         self.elements = elements
         self.keys = [v["name"].lower() for v in self.elements]
 
+    def _parse_colorcode(self, s):
+        """ Allow hex and oct values for colors """
+        s = s.strip()
+        if not s:
+            raise ValueError('Foreground and Background must be numeric')
```
Comment (alexp1917) [inline fpdf/template.py:81]: maybe consider putting a different error message as s is "not empty" here, not "not numeric". (unless i am missing something?)

File: fpdf/template.py (hunk 1 above)
Comment (gmischler) [reply fpdf/template.py:93]: Everything got stripped in the original version, so I just left it that way in order not to change the behaviour. The alternative would be to just use `str` as a NOP.

File: fpdf/template.py (hunk 2 above)
Comment (gmischler) [reply fpdf/template.py:81]: I phrased it in this way to inform the user about what's supposed to go there. "Not empty" doesn't give that information.
The documentation could also be more specific about this, but fleshing that out is a project for another time...

File: fpdf/template.py [hunk 3]
```diff
@@ -74,26 +73,42 @@ def load_elements(self, elements):
...
-            "foreground",
-            "background",
-            "align",
-            "text",
-            "priority",
-            "multiline",
+        def varsep_float(s):
+            """Convert to float with given decimal seperator"""
+            # glad to have nonlocal scoping...
+            return float(s.replace(decimal_sep, '.'))
```
Comment (gmischler) [inline fpdf/template.py:93]: I hadn't payed enough attention to decimal seperator replacement at first.
It previously was done on any field where the content didn't start with a ' (single quote).
Those are not at all garanteed to be present in text fields, so we need to restrict replacement to actual float value fields.
Fortunately a nested function sees the surrounding scope where the seperator is defined, which makes this easy.

File: test/template/test_template.py [hunk 4]
```diff
@@ -28,6 +28,7 @@ def test_template_nominal_hardcoded(tmp_path):
             "align": "I",
             "text": "logo",
             "priority": 2,
+            "multiline": 0,
```
Comment (Lucas-C) [inline test/template/test_template.py:31]: Did `multiline` become mandatory?

**Comment by Lucas-C** [comment]
Thank you for your contribution @gmischler !

I removed `util.try_to_type` & updated the CHANGELOG.md to mention it:
https://github.com/PyFPDF/fpdf2/blob/master/CHANGELOG.md#244---not-released-yet

**Comment by Lucas-C** [comment]
@all-contributors-bot please add @gmischler for code

**Comment by Lucas-C** [comment]
@all-contributors please add @gmischler for code

**Pull request merged**